src/test/java/com/google/api/codegen/testsrc/libraryproto/library_gapic.yaml

```diff
@@ -328,10 +328,33 @@ interfaces:
       parameters:
         defaults:
         - name="Unreal book"
+    - id: test_on_success_map
+      title: "test maps in response handling"
+      parameters:
+        defaults:
+        - name="test_on_success_map"
+      on_success:
+        - define: string_key_val = $resp.map_list_value_value{key}
```

Comment:
Where is `key` defined? Either I'm missing it, or this should fail.

Reply:
`key` is the literal of the actual name of the key in the map. Gapic-generator will try to parse  whatever is between the curly braces based on the key type. If [it can't](https://github.com/googleapis/gapic-generator/pull/2588/files#diff-1099a27774a5a8cfd19623070e23e508R409), it will fail.

For this specific behavior, the type of the key of the map is `string`, so gapic-generator will try to parse `key` as a string.

IMO gapic-generator is pretty liberal when it comes to parse literals from yaml configs. I think it's OK though, as long as the literals are rendered correctly according to the language-idiomatic syntax and their types.

That said, I am also OK to make the configuration more conservative (e.g., strings have to be quoted, etc) if y'all think this will benefit us more.

Reply:
I think in this instance we should require quotes here. Otherwise it's easier to confuse it for a variable name (which is what I assumed).

My thinking is this:
- an entire yaml  value can be quoted or not, as per yaml rules, etc.
- we should generally be stricter about subparts of a yaml value that we parse, so that strings should be quoted, variable names should be lower_camel, etc.

Reply:
SG. I'll check it here.